hexrd/ui/mask_manager_dialog.py

```diff
@@ -388,3 +400,12 @@ def show_masks(self):
             axis.imshow(HexrdConfig().raw_masks_dict[det])
         fig.canvas.draw_idle()
         fig.show()
+
+    def hide_masks(self, checked):
+        if checked:
+            HexrdConfig().visible_masks.clear()
+        else:
+            HexrdConfig().visible_masks = list(self.masks.keys())
```

Comment:
It looks to me that `HexrdConfig().visible_masks` is paired with the checkboxes in the table. And since we are trying to make those checkboxes and this checkbox separate, I think the `Hide all masks` checkbox should not modify the same variables.

However, to make the internal code design simpler, I think for now it might be easier to just add two buttons: `Hide all masks` and `Show all masks`. These would modify `HexrdConfig().visible_masks`, and then we would update all checkboxes in the table to reflect their new state.

The only reasons why a separate checkbox would be preferable are:

1. It is only one widget (a checkbox) instead of two buttons (so takes up less space in the UI)
2. It could allow global control of mask visibilities that is separate from individual control (the checkboxes in the table). But I think if we wanted this, we should add a separate variable for the global mask visibilities (similar to `HexrdConfig().show_overlays`), and save that in the state as well.

Instead of the checkbox, I think buttons are fine for now unless someone asks for the checkbox.

Reply:
Fixed